src/main/java/course_project/Main.java

```diff
@@ -0,0 +1,8 @@
+package course_project;
+
+public class Main {
+
+    public static void main(String[] args)  {
```

Comment:
Good job! Nice interface, a lot of features (such as automatic ship placement, different game modes), comfortable to play, informative messages. I like the architecture, the code is readable and understandable. You show good knowledge of Java Core (missing Java 8 and some concepts of OOP). 

To make it better: add automatic miss marks placement around a destroyed ship. Also, your code has a lot of static. It seems logical that if you don't care about having multiple objects of a class in your app (like Ship), then mark it static, but actually in this case it should be a singleton, and still dynamic. Static is rarely used in Java, mostly for constants and utils. And mostly your methods are void, which is also rare. 

Approved!